src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -884,38 +1035,44 @@ RelationalDBInterfaceSqliteImp::getOldestAccountTxs(
             .first;
     }
 
-    /* else use shard databases */
-    AccountTxs ret;
-    AccountTxOptions opt = options;
-    int limit_used = 0;
-    iterateTransactionForward(
-        opt.minLedger ? seqToShardIndex(opt.minLedger)
-                      : std::optional<std::uint32_t>(),
-        [&](soci::session& session, std::uint32_t index) {
-            if (opt.maxLedger && index > seqToShardIndex(opt.maxLedger))
-                return false;
-            auto [r, total] = ripple::getOldestAccountTxs(
-                session, app_, ledgerMaster, opt, limit_used, j_);
-            ret.insert(ret.end(), r.begin(), r.end());
-            if (!total)
-                return false;
-            if (total > 0)
-            {
-                limit_used += total;
-                opt.offset = 0;
-            }
-            else
-            {
-                total = ~total;
-                if (opt.offset <= total)
+    /* else use shard databases, if available */
+    if (existsShardTransaction())
+    {
+        AccountTxs ret;
+        AccountTxOptions opt = options;
+        int limit_used = 0;
+        iterateTransactionForward(
+            opt.minLedger ? seqToShardIndex(opt.minLedger)
+                          : std::optional<std::uint32_t>(),
+            [&](soci::session& session, std::uint32_t index) {
+                if (opt.maxLedger && index > seqToShardIndex(opt.maxLedger))
+                    return false;
+                auto [r, total] = ripple::getOldestAccountTxs(
+                    session, app_, ledgerMaster, opt, limit_used, j_);
+                ret.insert(ret.end(), r.begin(), r.end());
+                if (!total)
+                    return false;
+                if (total > 0)
+                {
+                    limit_used += total;
                     opt.offset = 0;
+                }
                 else
-                    opt.offset -= total;
-            }
-            return true;
-        });
+                {
+                    total = ~total;
```

Comment:
Bitwise invert? That looks like a bug. If it's not a bug it definitely needs a comment (here and the other places doing `total = ~total`.

Reply:
Added comments on it.

Reply:
Thanks for adding the comment, but that still doesn't clarify why it was done like this.I would have expected `-number` not `~number`.

It looks in all these function, the code looks like:
```
        if (!total && limit_used)
        {
             // ....
             total = ~total
        }
```
If that's true, then if we used negative instead of bitwise invert, then `0` means none processed and none skipped, positive means num processed, and negative means num skipped. Is there a reason `~` was chosen over `-`?

The only reason I could think of, is if we needed to distinguish between 0 processed and 0 skipped, but that doesn't appear to be the case. 

This isn't part of this PR, so no change needed, tho I'd still like to understand why `~` was chosen.